src/v/cluster/log_eviction_stm.cc

```diff
@@ -122,15 +162,8 @@ ss::future<> log_eviction_stm::monitor_log_eviction() {
         try {
             _storage_eviction_offset = co_await _raft->monitor_log_eviction(
               _as);
-            const auto max_collectible_offset
-              = _raft->log().stm_manager()->max_collectible_offset();
-            const auto next_eviction_offset = std::min(
-              max_collectible_offset, _storage_eviction_offset);
```

Comment:
this seems important, but now it doesn't seem like we consult max collectible offset in this. is it happening somewhere else?

mm, i guess maybe now it is happening over when index_lb/truncation_point is examined in do_write_raft_snapshot